scalac-scoverage-plugin/src/main/scala/scoverage/plugin.scala

```diff
@@ -35,16 +34,23 @@ class ScoveragePlugin(val global: Global) extends Plugin {
         options.dataDir = opt.substring("dataDir:".length)
       } else if (opt.startsWith("extraAfterPhase:") || opt.startsWith("extraBeforePhase:")) {
         // skip here, these flags are processed elsewhere
-      } else {
+      } else if (opt.startsWith("useEnvironment:")) {
+        options.useEnvironment = opt.substring("useEnvironment:".length).toBoolean
+      } else{
         error("Unknown option: " + opt)
       }
     }
-    if (!opts.exists(_.startsWith("dataDir:")))
-      throw new RuntimeException("Cannot invoke plugin without specifying <dataDir>")
+
+    if (opts.exists{p: String => p.contains("useEnvironment") && p.contains("false")} || (!opts.exists(_.startsWith("useEnvironment")))) {
```

Comment:
Is this meant to replace `dataDir` in the future?

Reply:
No, this is just a check that if the option to write to the class is either set to false `p.contains("useEnvironment") && p.contains("false")` or is not present `(!opts.exists(_.startsWith("useEnvironment"))`, then make sure then `dataDir` attribute is given by the user. 

Overriding dataDir takes place in `def setOptions(options: ScoverageOptions): Unit`